ext/test/http/curl_http_test.cc

```diff
@@ -248,16 +252,46 @@ TEST_F(BasicCurlHttpTests, CurlHttpOperations)
   std::multimap<std::string, std::string, curl::curl_ci> m1 = {
       {"name1", "value1_1"}, {"name1", "value1_2"}, {"name2", "value3"}, {"name3", "value3"}};
   curl::Headers headers = m1;
-  curl::HttpOperation http_operations1(http_client::Method::Head, "/get", handler, headers, body,
-                                       true);
+  curl::HttpOperation http_operations1(http_client::Method::Head, "/get", handler,
+                                       curl::RequestMode::Async, headers, body, true);
   http_operations1.Send();
 
-  curl::HttpOperation http_operations2(http_client::Method::Get, "/get", handler, headers, body,
-                                       true);
+  curl::HttpOperation http_operations2(http_client::Method::Get, "/get", handler,
+                                       curl::RequestMode::Async, headers, body, true);
   http_operations2.Send();
 
-  curl::HttpOperation http_operations3(http_client::Method::Get, "/get", handler, headers, body,
-                                       false);
+  curl::HttpOperation http_operations3(http_client::Method::Get, "/get", handler,
+                                       curl::RequestMode::Async, headers, body, false);
   http_operations3.Send();
   delete handler;
 }
+
+TEST_F(BasicCurlHttpTests, SendGetRequestSync)
+{
+  received_requests_.clear();
+  curl::SessionManager session_manager;
+
+  auto session = session_manager.CreateSession("127.0.0.1", HTTP_PORT);
+  auto request = session->CreateRequest();
+  request->SetUri("get/");
+  http_client::SessionState session_state;
+  auto response = session->SendRequestSync(session_state);
```

Comment:
What happens in his test if destination server is unreachable? Does `session->SendRequestSync` return an object? It appears like it'd return `nullptr` in this case, and subsequent code would crash?

Reply:
I was expecting the caller to check the status first, and accordingly read the response. But that won't be correct to assume.  Have modified the code to return empty response in case of errors , and caller should read response only when `session_state` is `Response`.